### PR TITLE
#6610 Fixed BILTerrainProvider with proxy

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
When a terrain provider needs to use proxy, the terrain provider fails because:
 - The URL was passed with the proxy, url encoded, and adding some parameters simply doesn't work
 - the proxy.properties do not contain the proper format to allow requests of type BIL

This PR 
- provides the proxy to the TerrainProvider, that will be used to create the final URLs of the tiles, after parsing the template and adding parameters.
- adds to `proxy.properties` the data format to allow

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#6610

**What is the current behavior?**

#<issue>

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/utils/cesium/BILTerrainProvider.js [hunk 1]
```diff
@@ -963,10 +950,11 @@ module.exports = function(Cesium) {
 					resultat.getHeightmapTerrainDataImage=function(x,y,level){
 						var retour;
 						if(!isNaN(x+y+level)){
-							var urlArray=templateToURL(resultat.URLtemplateImage(x,y,level),x, y, level,provider);
+							var url=templateToURL(resultat.URLtemplateImage(x,y,level),x, y, level,provider);
 							var limitations={highest:resultat.highest,lowest:resultat.lowest,offset:resultat.offset};
+                            var proxy = resultat.proxy || { getURL: v => v } ;
 							var hasChildren = terrainChildrenMask(x, y, level,provider);
-							var promise = Cesium.throttleRequestByServer(urlArray,Cesium.loadImage);
+                            var promise = Cesium.throttleRequestByServer(proxy.getURL(urlArray),Cesium.loadImage);
```
Comment (mbarto) [inline web/client/utils/cesium/BILTerrainProvider.js:957]: Is this correct? You renamed urlArray to url above, so I would expect url to be used here

**Review by mbarto — COMMENTED** [review]
LGTM, just one doubt, look at the related comment

File: web/client/utils/cesium/BILTerrainProvider.js (hunk 1 above)
Comment (offtherailz) [reply web/client/utils/cesium/BILTerrainProvider.js:957]: Probably I missed to save before committing. Thank you, nice catch. Tomorrow I'll try again this fix and merge this suggestion

```suggestion
                            var promise = Cesium.throttleRequestByServer(proxy.getURL(url),Cesium.loadImage);
```

**Comment by ElenaGallo** [comment]
@offtherailz Opening the context, without the 3D active, the elevation does not work

*Steps to reproduce*
- Open [this context]((https://dev.mapstore2.geo-solutions.it/mapstore/#/context/dem3d)
- Click on Show coordinate on footer
- Hover over the layer on map

*Current Result*
The layer elevation is not displayed in the footer

**Comment by offtherailz** [comment]
Thank you @ElenaGallo , I found out the problem, going to do a second PR to fix.
Because of this issue i noticed another thing that can be removed to improve performances and stability.

**Pull request merged**